pywinauto/linux/application.py

```diff
@@ -0,0 +1,171 @@
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Linux Application class"""
+
+import os.path
+import time
+import subprocess
+import shlex
+
+from ..backend import registry
+from ..base_application import AppStartError, ProcessNotFoundError, AppNotConnected, BaseApplication
+
+
+class Application(BaseApplication):
+
+    def __init__(self, backend="atspi", allow_magic_lookup=True):
+        """
+        Initialize the Application object
+
+        * **backend** is a name of used back-end (values: "atspi").
+	* **allow_magic_lookup** whether attribute access must turn into
+		child_window(best_match=...) search as fallback
+        """
+        self.process = None
+        self.xmlpath = ''
+
+        self._proc_descriptor = None
+        self.match_history = []
+        self.use_history = False
+        self.actions = None # TODO Action logger for linux
+        if backend not in registry.backends:
+            raise ValueError('Backend "{0}" is not registered!'.format(backend))
+        self.backend = registry.backends[backend]
+        self.allow_magic_lookup = allow_magic_lookup
+
+    def start(self, cmd_line, timeout=None, retry_interval=None,
+              create_new_console=False, wait_for_idle=True, work_dir=None):
+        command_line = shlex.split(cmd_line)
+        try:
+            process = subprocess.Popen(command_line, shell=create_new_console)
+        except Exception as exc:
+            # if it failed for some reason
+            message = ('Could not create the process "%s"\n'
+                       'Error returned by CreateProcess: %s') % (cmd_line, str(exc))
+            raise AppStartError(message)
+        self._proc_descriptor = process
+        self.process = process.pid
+        return self
+
+    def connect(self, **kwargs):
+        """Connect to an already running process
+
+        The action is performed according to only one of parameters
+
+        :param process: a process ID of the target
+        :param path: a path used to launch the target
+
+        .. seealso::
+
+           :func:`pywinauto.findwindows.find_elements` - the keyword arguments that
+           are also can be used instead of **process**, **handle** or **path**
```

Comment:
`handle` is not mentioned above in Linux implementation. It should be removed here as well.